File: Sorting algorithms/quick.py
def quick(arr):
    def partitioning(arr):
        i=0
        for j in range(len(arr)-1):
            if arr[j]<=arr[len(arr)-1]:
                arr[j],arr[i]=arr[i],arr[j]
                i+=1
            else: continue
        arr[i],arr[len(arr)-1]=arr[len(arr)-1],arr[i]
        print("2 array:", arr)
        print("The I is:", i,"element. It is:", arr[i])
        print(arr[i:    ])
        return i


    return partitioning(arr)

arr=[9,8,7,6,5,4,3,2,1]
#arr=[67,23,12,6,421,1,0,7,55,24,12,123,35,3,99,28]
print("Array length: ", len(arr)-1)
print("1 array:", arr)
print(quick(arr))

File: Sorting algorithms/quick.js
function quick(arr,l,h){
  function parti(arr,l,h){
    var j=l;
    var piv = h;
    for(i=l;i<piv;i++){
      if(arr[i]<=arr[piv]){
        [[arr[i]],[arr[j]]]=[[arr[j]],[arr[i]]];
        j++;
      }
      else{
        continue;
      }
    }
    [[arr[piv]],[arr[j]]]=[[arr[j]],[arr[piv]]];
    console.log("The array is :"+arr);
    console.log("The pivot is :"+arr[j]);
    return j;
  }
  function sort(arr,l,h){
    if(l<h){
      var p = parti(arr,l,h);
      sort(arr,p+1,h);
      sort(arr,l,p-1);
    }
  }
  sort(arr,l,h);
  return arr;
}

function getRandomInt(min, max) {
  return Math.floor(Math.random() * (max - min)) + min;
}
var arr = [456,23,888,123,34,33,1,0,45,99,320,50];
var l=0;
var h = arr.length-1;
console.log("Here is unsort array: "+arr);
console.log("Here is sorted array: "+quick(arr,l,h));
